therapy_app/client.py

```diff
@@ -0,0 +1,175 @@
+    permission_classes = [IsClient]
+
+    def get(self, request, id1, format=None):
+        """
+        This method will map a therapist that is not actively mapped to the requesting client.
+        """
+        if models.User.objects.filter(is_deleted=False, type='Therapist', pk=id1).exists():
+            try:
+                therapist = models.User.objects.filter(is_deleted=False, type='Therapist').exclude(
+                        Q(mapping_therapist__client=request.user) & Q(mapping_therapist__is_deleted=False)).get(pk=id1)
+                new_mapping = models.Mapping.objects.create(client=request.user,
+                                                            therapist=therapist)
+                return Response({'message': 'Mapping has been created',
+                                 'therapist_id': therapist.pk,
+                                 'client_id': request.user.pk}, status=status.HTTP_200_OK)
+            except Exception as e:
+                return Response({'message': 'The Therapist is already mapped.'}, status=status.HTTP_200_OK)
+        else:
+            return Response({'message': 'No such therapist exists'}, status=status.HTTP_404_NOT_FOUND)
+
+    def delete(self, request, id1, format=None):
+        """
+        This method will remove the mapping of a therapist and a requesting client.
+        """
+        if models.User.objects.filter(is_deleted=False, type='Therapist', pk=id1).exists():
+            try:
+                mapping = models.Mapping.objects.get(is_deleted=False, client=request.user, therapist__id=id1)
+                mapping.is_deleted = True
+                mapping.save()
+                return Response({"message": "Mapping has been removed.",
+                                 "therapist_id": id1,
+                                 "client_id": request.user.id}, status=status.HTTP_200_OK)
+            except Exception as e:
+                return Response({'data': 'There does not exist such an active mapping.'},
+                                status=status.HTTP_417_EXPECTATION_FAILED)
+        else:
+            return Response({'message': 'No such therapist exists'}, status=status.HTTP_404_NOT_FOUND)
+
+
+class MappedTherapistsView(APIView):
+    """
+    This APIView takes GET request only to list mapped therapists to clients.
+    This only shows those therapists that are actively mapped to the requesting client.
+    """
+    authentication_classes = (CsrfExemptSessionAuthentication,)
+    permission_classes = [IsClient]
+
+    def get(self, request, format=None):
+        """
+        This method will show therapists that are actively mapped to the requesting client.
+        """
+        mappings = models.Mapping.objects.filter(is_deleted=False, client=request.user)
+        if mappings:
+            serialized = serializers.MappingSerializer(mappings, many=True)
+            return Response(serialized.data, status=status.HTTP_200_OK)
```

Comment:
Its always good to have a fixed top level structure in your requests - for ex : 
`{
"status": "OK",
"data" : {...}
}`
and 
`{
"status": "ERROR",
"message": "<error string>"
}`